### PR TITLE
docs: Update the docs for Helm mode Cilium CLI

### LINKED ISSUE #1624: Update CLI documentation for Helm mode
For the most part, documentation that relates to the Cilium CLI remains unchanged after the implementation of Helm mode.

There are a few notable changes that will need to be made:

- [ ] Initial Cilium/Documentation pass which updates references to the new Helm-mode version and replaces all instances of `cilium-cli` commands with their Helm mode equivalents. All instances where removed flags are used should be updated to use their `--helm-set` equivalents
- [ ] New Doc: All removed flags should be documented with alternative usage using `--helm-*` flags
- [ ] The nuances of `--reuse-values` and `--reset-values` should be clearly documented. 

When using `cilium upgrade` a user may need to render their existing values in order to avoid having many values reset. A general solution for this is a hotly discussed topic in the upstream Helm community. As part of this documentation, we should add comments to the codebase as appropriate to explain the situation as well.
    - A general recommendation to avoid the strange behavior of `--reuse-values` is to perform any upgrade in two steps. First, upgrade the chart version with no values provided. Then, upgrade any values with `--reuse-values` provided.
    - See these upstream issues:
      https://github.com/helm/helm/issues/8085
      https://github.com/helm/helm/pull/9653
      https://github.com/helm/helm/pull/9426
    - This blog post explains the issue well https://shipmight.com/blog/understanding-helm-upgrade-reset-reuse-values
    
helm-upgrade-article table

### PR DESCRIPTION
**Pull request by michi-covalent**
- Consistently use the --set flag.
- Replace --helm-auto-gen-values with --dry-run-helm-values.
- Set kubeProxyReplacement to "true" instead of "strict".

Ref: #26430
Fixes: https://github.com/cilium/cilium-cli/issues/1624

### CODE REVIEW

File: Documentation/network/clustermesh/aks-clustermesh-prep.rst
Comment (jschwinger233) [inline Documentation/network/clustermesh/aks-clustermesh-prep.rst:93]: This `azure.resourceGroup` seems to be missed in the [Documentation/helm-values.rst](https://github.com/cilium/cilium/blob/main/Documentation/helm-values.rst)

**Review by jschwinger233 — APPROVED** [review]

File: Documentation/network/clustermesh/aks-clustermesh-prep.rst
Comment (michi-covalent) [reply Documentation/network/clustermesh/aks-clustermesh-prep.rst:93]: yeah some of these azure settings are not documented 😞 https://github.com/cilium/cilium/blob/c068a443049b5dedb34608069df4b1721b1e4669/install/kubernetes/cilium/values.yaml#L329-L335

File: Documentation/installation/k8s-install-migration.rst
Comment (giorio94) [inline Documentation/installation/k8s-install-migration.rst]:
```suggestion
        $ cilium install --values values-migration.yaml --dry-run-helm-values > values-initial.yaml
```

Nit. I personally slightly prefer the long form, as a bit more self-explanatory (if accepted, the same would apply to the other occurrences). WDYT?

File: Documentation/operations/performance/tuning.rst
Comment (giorio94) [inline Documentation/operations/performance/tuning.rst]: When in a code-block (differently from a parsed-literal block as below), `\` does not need to be escaped.

File: Documentation/network/servicemesh/gateway-api/installation.rst
Comment (giorio94) [inline Documentation/network/servicemesh/gateway-api/installation.rst]: Same here.

File: Documentation/security/network/encryption-ipsec.rst
Comment (giorio94) [inline Documentation/security/network/encryption-ipsec.rst]: Same here (and the subsequent occurrences).

File: Documentation/network/clustermesh/aks-clustermesh-prep.rst
Comment (giorio94) [inline Documentation/network/clustermesh/aks-clustermesh-prep.rst]:
```suggestion
            --set ipam.operator.clusterPoolIPv4PodCIDRList='{10.10.0.0/16}'
```

Seems to work better to me (especially in case multiple CIDRs are specified). For instance, `...CIDRList='{10.10.0.0/16,10.20.0.0/16}'` leads to 

```yaml
ipam:
  mode: kubernetes
  operator:
    clusterPoolIPv4PodCIDRList:
    - 10.10.0.0/16
    - 10.20.0.0/16
```

The same applies also to the one below.

**Review by giorio94 — COMMENTED** [review]
Thanks @michi-covalent for tackling this!

A couple of additional aspects:
* I wonder if it could make sense to add a callout to the `cli-download.rst` (or somewhere else) to explicitly state that these guides require Cilium CLI v0.15 or subsequent, and users should upgrade it if appropriate. Otherwise users that already have the CLI installed might be puzzled about why it is not working as expected.
* Similarly, I wonder if it could make sense to explicitly state that using the new CLI against an installation performed with older CLI versions is not supported, and can lead to errors/issues.
* There seems to be still a couple of leftover occurrences of `--helm-set` in `installation/k3s.rts` and `servicemesh/installation.rst`.

File: Documentation/installation/k8s-install-migration.rst
Comment (michi-covalent) [reply Documentation/installation/k8s-install-migration.rst]: updated 📝

**Comment by michi-covalent** [comment]
> I wonder if it could make sense to add a callout to the cli-download.rst (or somewhere else) to explicitly state that these guides require Cilium CLI v0.15 or subsequent, and users should upgrade it if appropriate. Otherwise users that already have the CLI installed might be puzzled about why it is not working as expected.
> 
> Similarly, I wonder if it could make sense to explicitly state that using the new CLI against an installation performed with older CLI versions is not supported, and can lead to errors/issues.

yeah let's do that. i'll move this to draft, and add notes about cilium-cli version / incompatibility with the classic mode 📝 

> There seems to be still a couple of leftover occurrences of --helm-set in installation/k3s.rts and servicemesh/installation.rst.

updated 📝 🚀

File: Documentation/operations/performance/tuning.rst
Comment (giorio94) [inline Documentation/operations/performance/tuning.rst]: Unfortunately in this case escaping is required because it is part of a parsed-literal block, which AFAIU it is a requirement to use the `CHART_RELEASE` fragment.

Comment (michi-covalent) [reply Documentation/operations/performance/tuning.rst]: tricky stuff

File: Documentation/network/clustermesh/aks-clustermesh-prep.rst
Comment (learnitall) [reply Documentation/network/clustermesh/aks-clustermesh-prep.rst:93]: Can we create a follow-up issue for documenting these options?

**Review by learnitall — COMMENTED** [review]
Thank you, looks great! LGTM pending a couple of things:

* @giorio94's comment on calling out compatibility concerns
* I think there are some files and changes that we missed. These are some grep calls in `Documentation/`

kube proxy replacement, strict vs true. Some of these may not be related to the cilium-cli changes however and may not need to be changed:

```
grep -i '.*proxy.*replacement.*strict' -n --exclude-dir '_*' -R .
./network/kubernetes/ipam-multi-pool.rst:32:   * ``--set kubeProxyReplacement=strict``
./network/kubernetes/kubeproxy-free.rst:105:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:158:    KubeProxyReplacement:   Strict	[eth0 (Direct Routing), eth1]
./network/kubernetes/kubeproxy-free.rst:383:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:447:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:519:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:550:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:566:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:595:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:631:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:667:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:808:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:869:     --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:962:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:974:By specifying ``kubeProxyReplacement=strict`` the native hostPort support is
./network/kubernetes/kubeproxy-free.rst:996:        --set kubeProxyReplacement=strict \\
./network/kubernetes/kubeproxy-free.rst:1113:- ``kubeProxyReplacement=strict``: When using this option, it's highly recommended
./network/kubernetes/kubeproxy-free.rst:1149:  The following Helm setup below would be equivalent to ``kubeProxyReplacement=strict``
./network/kubernetes/kubeproxy-free.rst:1200:    KubeProxyReplacement:   Strict	[eth0 (DR)]
./network/kubernetes/kubeproxy-free.rst:1301:Cilium's eBPF kube-proxy replacement restricts access from outside (e.g. external
./network/kubernetes/kata.rst:81:   :ref:`kube-proxy replacement <kubeproxy-free>` in strict mode. These
./network/egress-gateway.rst:93:               --set kubeProxyReplacement=strict \\
./network/egress-gateway.rst:103:            kube-proxy-replacement: strict
./network/l2-announcements.rst:42:               --set kubeProxyReplacement=strict \\
./network/l2-announcements.rst:52:            kube-proxy-replacement: strict
./network/l2-announcements.rst:57:* Kube Proxy replacement mode must be enabled and set to ``strict`` mode, see :ref:`kubeproxy-free` for details.
./network/l2-announcements.rst:292:               --set kubeProxyReplacement=strict \\
./network/l2-announcements.rst:304:            kube-proxy-replacement: strict
./operations/upgrade.rst:208:      kubeProxyReplacement: "strict"
./operations/performance/tuning.rst:91:             --set kubeProxyReplacement=strict
./operations/performance/tuning.rst:148:             --set kubeProxyReplacement=strict
./operations/performance/tuning.rst:260:             --set kubeProxyReplacement=strict
./operations/performance/tuning.rst:308:             --set kubeProxyReplacement=strict
./operations/troubleshooting_servicemesh.rst:23: #. Validate that the ``kubeProxyReplacement`` is set to either partial or strict.
./operations/troubleshooting_servicemesh.rst:31:        KubeProxyReplacement:    Strict   [eth0 192.168.49.2]
./configuration/per-node-config.rst:65:nodes with ``io.cilium.migration/kube-proxy-replacement: strict``
./configuration/per-node-config.rst:79:        kubectl -n kube-system patch daemonset kube-proxy --patch '{"spec": {"template": {"spec": {"affinity": {"nodeAffinity": {"requiredDuringSchedulingIgnoredDuringExecution": {"nodeSelectorTerms": [{"matchExpressions": [{"key": "io.cilium.migration/kube-proxy-replacement", "operator": "NotIn", "values": ["strict"]}]}]}}}}}}}'
./configuration/per-node-config.rst:90:          name: kube-proxy-replacement-strict
./configuration/per-node-config.rst:94:              io.cilium.migration/kube-proxy-replacement: strict
./configuration/per-node-config.rst:96:            kube-proxy-replacement: strict
./configuration/per-node-config.rst:106:        kubectl label node $NODE --overwrite 'io.cilium.migration/kube-proxy-replacement=strict'
./configuration/per-node-config.rst:133:        cilium config set --restart=false kube-proxy-replacement strict
./configuration/per-node-config.rst:135:        kubectl -n kube-system delete ciliumnodeconfig kube-proxy-replacement-strict
```

--set over --helm-set:

```
grep -i 'helm-set' -n --exclude-dir '_*' -R .
./network/clustermesh/clustermesh.rst:56: * ``cilium install`` option ``--helm-set ipv4NativeRoutingCIDR=10.0.0.0/9``
./network/clustermesh/clustermesh.rst:107: * Cilium CLI install options (*helm* mode) ``--helm-set cluster.name`` and ``--helm-set cluster.id``
```

**Comment by michi-covalent** [comment]
> kube proxy replacement, strict vs true. Some of these may not be related to the cilium-cli changes however and may not need to be changed:

yeah let's deal with kpr values in a separate PR if necessary. summoning @brb to check if we need to update all the pages to use `true` instead of `strict` 🍅 

> ```
> grep -i 'helm-set' -n --exclude-dir '_*' -R .
> ./network/clustermesh/clustermesh.rst:56: * ``cilium install`` option ``--helm-set ipv4NativeRoutingCIDR=10.0.0.0/9``
> ./network/clustermesh/clustermesh.rst:107: * Cilium CLI install options (*helm* mode) ``--helm-set cluster.name`` and ``--helm-set cluster.id``
> ```

these are addressed in https://github.com/cilium/cilium/pull/26608 ✅

**Comment by michi-covalent** [comment]
- cilium-cli installation instructions: https://deploy-preview-26606--docs-cilium-io.netlify.app/gettingstarted/k8s-install-default.html#install-the-cilium-cli
- upgrade guide: https://deploy-preview-26606--docs-cilium-io.netlify.app/operations/upgrade.html#cilium-cli

File: Documentation/network/clustermesh/aks-clustermesh-prep.rst
Comment (michi-covalent) [reply Documentation/network/clustermesh/aks-clustermesh-prep.rst:93]: opened https://github.com/cilium/cilium/issues/26654

**Review by mhofstetter — APPROVED** [review]

File: Documentation/installation/cli-download.rst
Comment (giorio94) [inline Documentation/installation/cli-download.rst:5]: Nit. It might make sense to add a link to the specific section of the upgrade notes.

**Review by giorio94 — APPROVED** [review]
/lgtm

File: Documentation/installation/cli-download.rst
Comment (michi-covalent) [reply Documentation/installation/cli-download.rst:5]: i'll update it before marking it ready to merge 📝

File: Documentation/operations/upgrade.rst
Comment (learnitall) [inline Documentation/operations/upgrade.rst]:
```suggestion
Upgrade Cilium CLI to `v0.15.0 <https://github.com/cilium/cilium-cli/releases/tag/v0.15.0>`_
```
Can be avoid use of please? It doesn't always translate well to other languages.

File: Documentation/operations/upgrade.rst
Comment (learnitall) [inline Documentation/operations/upgrade.rst]:
```suggestion
or later to switch to `Helm installation mode <https://github.com/cilium/cilium-cli#helm-installation-mode>`_
```

File: Documentation/operations/upgrade.rst
Comment (learnitall) [inline Documentation/operations/upgrade.rst]:
```suggestion
to install and manage Cilium v1.14. Classic installation mode is **not**
```

**Review by learnitall — CHANGES_REQUESTED** [review]
A couple of nits, but otherwise looks good! Since I am new to reviewing docs, let's also have another docs-structure team member take a look.

**Review by qmonnet — APPROVED** [review]
Looks all in order to me once the feedback from Marco and Ryan is addressed :+1:.

Skimming through the discussion: the other values for KPR should have been updated with https://github.com/cilium/cilium/pull/26577.

File: Documentation/installation/cli-download.rst
Comment (michi-covalent) [reply Documentation/installation/cli-download.rst:5]: ✨ https://deploy-preview-26606--docs-cilium-io.netlify.app/gettingstarted/k8s-install-default.html#install-the-cilium-cli ✨

**Review by learnitall — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**
